src/helpers.rs

```diff
@@ -72,6 +72,34 @@ pub(crate) fn parse_field_attributes(attributes: &[syn::Attribute]) -> HashMap<S
     attrs
 }
 
+// Returns true if union has the repr(C) attribute set.
+pub(crate) fn check_union_layout(attributes: &[syn::Attribute]) -> bool {
```

Comment:
There seems to be some common logic between this method and `parse_field_attributes`. Is there any way to avoid duplicating it ?

Also a nit: I would rename this fn to something like `has_C_layout` to make it more generic.

Reply:
`parse_field_attributes` builds and returns a hashtable of name/value while the new code just scans for repr(C) and returns true or false. Indeed the scanning part is common. I will look for ways to deduplicate it.

I will rename that function as u suggested which is more explicit, not generic IMO.

I'll ping you once I update.

Reply:
I've deduped the code and renamed function.